tiny-postgresql-event-store/src/main/kotlin/jp/veka/query/insert/OnDuplicateKeyUpdateInsertQuery.kt

```diff
@@ -0,0 +1,32 @@
+package jp.veka.query.insert
+
+import jp.veka.query.BasicQuery
+import jp.veka.query.exception.InvalidQueryStateException
+import java.sql.Connection
+
+class OnDuplicateKeyUpdateInsertQuery(schema: String, relation: String)
+    : BasicQuery<OnDuplicateKeyUpdateInsertQuery>(schema, relation) {
+    private val duplicateKeyUpdateColumns: MutableList<String> = mutableListOf()
+    fun onDuplicateKeyUpdateValues(vararg columns: String) : OnDuplicateKeyUpdateInsertQuery {
+        for (column in columns) {
+            duplicateKeyUpdateColumns.add(column)
+        }
+        return this
+    }
+    override fun execute(connection: Connection) : Boolean {
+        return connection.prepareStatement(
+            "insert into ${schema}.${relation} (${columns.joinToString(",")}) values (${ values.joinToString(",")})" +
+                "on duplicate key update" +
```

Comment:
1. Have you tested that it is working for postgres?
2. I'm not sure even if it works that it works as expected here. We need some kind of CAS operation like "set X=2 where Y==3"